File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.rb
# @param {Integer} num
# @param {Integer} k
# @return {Integer}
def divisor_substrings(num, k)
    count=0
    nums=num.digits.reverse
    (nums.size-(k-1)).times do |i|
        window = nums[i, k]
        p windowx = window.join.to_i
        if windowx.nonzero?
            count+=1 if (num % windowx) == 0
            p "#{(num % windowx == 0)} | #{count}"
        end
    end
    count
end

File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:  
        count=0     
        nums = str(num)
        for i in range(len(nums)-(k-1)):
            window = int(nums[i:k+i])
            if window != 0 and (num % window) == 0:
                count+=1

        return count